xarray/core/variable.py

```diff
@@ -943,8 +943,13 @@ def __setitem__(self, key, value):
 
         See __getitem__ for more details.
         """
+        from xarray.core.dataarray import DataArray
+
         dims, index_tuple, new_order = self._broadcast_indexes(key)
 
+        if isinstance(value, DataArray):
```

Comment:
This feels a bit unpythonic (not duck-typing friendly), but there are lot of other `isinstance` checks in this module, so I didn't think about it too much.